<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>578</width>
    <height>425</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CPU &amp; OS Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Generación de programas</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>541</width>
          <height>311</height>
         </rect>
        </property>
        <property name="title">
         <string>Gestión de código fuente y código máquina</string>
        </property>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>20</y>
           <width>191</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="lineWidth">
          <number>1</number>
         </property>
         <property name="text">
          <string>Archivos tipo .LM</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>20</y>
           <width>191</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="text">
          <string>Archivos tipo .ASIM</string>
         </property>
        </widget>
        <widget class="QListWidget" name="listWidget_LM">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>40</y>
           <width>191</width>
           <height>131</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
background-color: rgb(255, 255, 204);
 border-style: outset;
 border-width: 1px;
}</string>
         </property>
        </widget>
        <widget class="QListWidget" name="listWidget_ASM">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>40</y>
           <width>191</width>
           <height>131</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
background-color: rgb(0, 204, 255);
 border-style: outset;
 border-width: 1px;
}</string>
         </property>
        </widget>
        <widget class="QWidget" name="verticalLayoutWidget">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>180</y>
           <width>131</width>
           <height>134</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="pushButton_20">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Carga en las listas de archivos según extensión"/>
            </property>
            <property name="text">
             <string>CARGAR ASIM</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_17">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Editar ASIM</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_14">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Nuevo ASIM</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_18">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Borrar ASIM</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="verticalLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>290</x>
           <y>180</y>
           <width>121</width>
           <height>100</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="pushButton_22">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string extracomment="Carga en las listas de archivos según extensión"/>
            </property>
            <property name="text">
             <string>CARGAR LM</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_21">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Ver LM</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_6">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>7</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Borrar LM</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QPushButton" name="pushButton_19">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>180</y>
           <width>121</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>GENERAR EJECUTABLE</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_15">
         <property name="geometry">
          <rect>
           <x>420</x>
           <y>180</y>
           <width>111</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="toolTip">
          <string extracomment="Crear el proceso y lo coloca en lista de espera para ejecución"/>
         </property>
         <property name="text">
          <string extracomment="Agrega a la lista de espera en la pestaña de simulación">PARA SIMULACION</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Simulación</string>
       </attribute>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>246</width>
          <height>286</height>
         </rect>
        </property>
        <property name="title">
         <string>Gestión de procesos</string>
        </property>
        <widget class="QListWidget" name="listWidget_Process">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>40</y>
           <width>181</width>
           <height>151</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>8</pointsize>
           <weight>9</weight>
           <italic>false</italic>
           <bold>false</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
	background-color: rgb(160, 210, 160);
    border-style: outset;
    border-width: 1px;
    border-color: black;
	font: 75 8pt &quot;Arial&quot;;
}</string>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="showDropIndicator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>200</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Crear Proceso (s)</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_2">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>230</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Muestra Consolas</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_5">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>230</y>
           <width>101</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Muestra Estados</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>181</width>
           <height>16</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <property name="text">
          <string>Ejecutables disponibles</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_8">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>200</y>
           <width>101</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Muestra Memoria</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_9">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>260</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Quitar de Lista</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="geometry">
         <rect>
          <x>275</x>
          <y>20</y>
          <width>261</width>
          <height>286</height>
         </rect>
        </property>
        <property name="title">
         <string>Control de la simulación</string>
        </property>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="geometry">
          <rect>
           <x>40</x>
           <y>80</y>
           <width>171</width>
           <height>181</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">QGroupBox {
background-color: rgb(255, 255, 204);
 border-style: outset;
 border-width: 1px;
}</string>
         </property>
         <property name="title">
          <string> Variables </string>
         </property>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>70</y>
            <width>141</width>
            <height>20</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Arial</family>
           </font>
          </property>
          <property name="text">
           <string>0 -- Tiempo Animación -- 10</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="line_RAM">
          <property name="geometry">
           <rect>
            <x>70</x>
            <y>155</y>
            <width>36</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>10000</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QSlider" name="hsTime">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>50</y>
            <width>131</width>
            <height>20</height>
           </rect>
          </property>
          <property name="toolTip">
           <string extracomment="Velocidad de animacaión"/>
          </property>
          <property name="statusTip">
           <string extracomment="Control de tiempo en simulación."/>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
          <property name="sliderPosition">
           <number>5</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QRadioButton" name="radioButton">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>20</y>
            <width>111</width>
            <height>17</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Arial</family>
           </font>
          </property>
          <property name="text">
           <string>Modo depuración</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>30</x>
            <y>155</y>
            <width>36</width>
            <height>16</height>
           </rect>
          </property>
          <property name="font">
           <font>
            <family>Arial</family>
           </font>
          </property>
          <property name="text">
           <string>RAM:</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="time">
          <property name="geometry">
           <rect>
            <x>65</x>
            <y>90</y>
            <width>31</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>5</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QSlider" name="hsRAM">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>130</y>
            <width>131</width>
            <height>20</height>
           </rect>
          </property>
          <property name="toolTip">
           <string extracomment="RAM del PC virtual"/>
          </property>
          <property name="statusTip">
           <string extracomment="Control de tiempo en simulación."/>
          </property>
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="pageStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>10000</number>
          </property>
          <property name="sliderPosition">
           <number>10000</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickInterval">
           <number>50</number>
          </property>
         </widget>
        </widget>
        <widget class="QPushButton" name="pushButton_3">
         <property name="geometry">
          <rect>
           <x>95</x>
           <y>30</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>DETENER</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>81</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>INICIAR</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_7">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>30</y>
           <width>81</width>
           <height>23</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>7</pointsize>
          </font>
         </property>
         <property name="text">
          <string>PAUSAR</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_Simula">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>60</y>
           <width>201</width>
           <height>16</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>9</pointsize>
           <weight>75</weight>
           <bold>true</bold>
           <underline>true</underline>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>578</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuOpciones">
    <property name="title">
     <string>Opciones</string>
    </property>
    <addaction name="actionSalir"/>
   </widget>
   <widget class="QMenu" name="menuAyuda">
    <property name="title">
     <string>Ayuda</string>
    </property>
    <addaction name="actionAcerca_de"/>
    <addaction name="actionArquitectura_MV"/>
   </widget>
   <addaction name="menuOpciones"/>
   <addaction name="menuAyuda"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAcerca_de">
   <property name="text">
    <string>Acerca de</string>
   </property>
  </action>
  <action name="actionSalir">
   <property name="text">
    <string>Salir</string>
   </property>
  </action>
  <action name="actionArquitectura_MV">
   <property name="text">
    <string>Arquitectura MV</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionSalir</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>salir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAcerca_de</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>acercade()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionArquitectura_MV</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>ayuda()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>salir()</slot>
  <slot>acercade()</slot>
  <slot>ayuda()</slot>
 </slots>
</ui>
